includes/class-sensei-quiz.php

```diff
@@ -1451,6 +1451,18 @@ public static function the_user_status_message( $quiz_id ) {
 
 		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- Output escaped above.
 		echo $message;
+
+		// show incomplete questions info message if there are incomplete questions and is preview.
+		if ( is_preview() ) {
+			$lesson_id          = Sensei()->quiz->get_lesson_id( $quiz_id );
+			$all_questions      = Sensei()->quiz->get_questions( Sensei()->lesson->lesson_quizzes( $lesson_id ) );
+			$filtered_questions = Sensei()->quiz->get_questions( Sensei()->lesson->lesson_quizzes( $lesson_id ), 'any', 'meta_value_num title', 'ASC', true );
```

Comment:
Instead of calling the `get_questions()` method twice, you could pass the results to `filter_out_incomplete_questions()` and check if there is a difference.

If you want to take things further, you could also make a `has_incomplete_questions(.$questions )` helper method that just checks if there are incomplete questions in a posts array.

You could also extract the logic that checks if a question is incomplete from the `filter_out_incomplete_questions` method, lets say `is_question_incomplete(.$question )`, and use that in `has_incomplete_questions` and `filter_out_incomplete_questions`. That way if you have 1k questions you don't filter them all out before knowing if there is an incomplete one.

Reply:
ok good point